app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt

```diff
@@ -25,43 +46,53 @@ class MessageViewHolder(
     }
 
     override fun bindViews(data: MessageUiModel) {
-        with(itemView) {
-            day_marker_layout.visibility = if (data.showDayMarker) {
-                day.text = data.currentDayMarkerText
-                View.VISIBLE
-            } else {
-                View.GONE
-            }
+        itemView.day_marker_layout.visibility = if (data.showDayMarker) {
+            itemView.day.text = data.currentDayMarkerText
+            View.VISIBLE
+        } else {
+            View.GONE
+        }
 
-            if (data.isFirstUnread) {
-                new_messages_notif.visibility = View.VISIBLE
-            } else {
-                new_messages_notif.visibility = View.GONE
-            }
+        if (data.isFirstUnread) {
+            itemView.new_messages_notif.visibility = View.VISIBLE
+        } else {
+            itemView.new_messages_notif.visibility = View.GONE
+        }
 
-            text_message_time.text = data.time
-            text_sender.text = data.senderName
-            text_content.text = data.content
-            image_avatar.setImageURI(data.avatar)
-            text_content.setTextColor(if (data.isTemporary) Color.GRAY else Color.BLACK)
+        itemView.text_message_time.text = data.time
+        itemView.text_sender.text = data.senderName
 
-            data.message.let {
-                text_edit_indicator.isVisible = !it.isSystemMessage() && it.editedBy != null
-                image_star_indicator.isVisible = it.starred?.isNotEmpty() ?: false
+        if (data.content is Spannable) {
+            val spans = data.content.getSpans(0, data.content.length, ImageSpan::class.java)
+            spans.forEach {
+                if (it.drawable is GifDrawable) {
+                    it.drawable.callback = this@MessageViewHolder
```

Comment:
I'm also kind unsure, but I think it's not needed, because the callback attaches to a drawable, which itself is attached to a Spannable inside the TextView, but as long as the view is recycled and a new text/spannable is entered the drawable would be wiped along with the spannable.

Reply:
Let's keep one eye on this after release.